### PR TITLE
Add Apache 2.0 license headers to source files

### LINKED ISSUE #17: Add Apache 2 license headers to repository files
In order to ensure all our Enarx repositories are properly licensed, we should ensure all repository files have a proper Apache 2 header added to them.

### PR DESCRIPTION
**Pull request by mbestavros**
Resolves #17.

### CODE REVIEW

**Review by frozencemetery — CHANGES_REQUESTED** [review]
Hmm, this isn't quite how I'd expect to see this.

Typically we'd expect a file - LICENSE or COPYING - at the top level of the repo with the full (Apache, if that's what we're going with) license text in it.  We don't have that already, which is an oversight - I think you should add that as part of this.

Having that file allows us to simplify the header on other files - you can do something like `/* Copyright 2019 the enarx contributors; see COPYING for details */` as a single line file-initial comment and have a much shorter boilerplate.  If a reference would help, you can look at [gssproxy](https://github.com/frozencemetery/gssproxy) for an example of what I mean here.

**Comment by npmccallum** [comment]
@frozencemetery Sometimes files get copied into other projects without source info. It is my preference to have *both* the license file and a full file header.

**Comment by mbestavros** [comment]
Updated with separate Apache 2.0 license.

File: src/header.rs [hunk 1]
```diff
@@ -1,3 +1,16 @@
 #![allow(dead_code)]
+// Copyright 2019 Red Hat
```
Comment (frozencemetery) [inline src/header.rs:1]: Should be above the `#![allow(dead_code)]` line.

File: src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,17 @@
 #![allow(unconditional_recursion)]
+// Copyright 2019 Red Hat
```
Comment (frozencemetery) [inline src/lib.rs:1]: Same here.

**Review by frozencemetery — CHANGES_REQUESTED** [review]
Alright, full header it is.

**Review by npmccallum — APPROVED** [review]

**Pull request merged**